modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/utils/SdkComponentsMinMuleVersionUtils.java

```diff
@@ -0,0 +1,655 @@
+    SdkComponent interfaceComponent = getEnforcedMinMuleVersion(interfaceType);
+    if (belongsToSdkPackages(interfaceType.getTypeName())) {
+      return interfaceComponent;
+    }
+    Optional<SdkComponent> superInterface = interfaceType.getImplementingInterfaces()
+        .map(SdkComponentsMinMuleVersionUtils::calculateInterfaceMinMuleVersion).reduce(SdkComponentsMinMuleVersionUtils::max);
+    superInterface.ifPresent(sdkComponent -> interfaceComponent.updateIfHigherMMV(sdkComponent,
+                                                                                  format("Interface %s has min mule version %s because it implements %s.",
+                                                                                         interfaceType.getName(),
+                                                                                         sdkComponent
+                                                                                             .getMinMuleVersion(),
+                                                                                         sdkComponent
+                                                                                             .getName())));
+    return interfaceComponent;
+  }
+
+  private static SdkComponent calculateSourceGenericsMinMuleVersion(List<Type> sourceGenerics) {
+    Type outputType = sourceGenerics.get(0);
+    Type attributesType = sourceGenerics.get(1);
+    return max(getEnforcedMinMuleVersion(attributesType), calculateOutputMinMuleVersion(outputType));
+  }
+
+  private static SdkComponent calculateOutputMinMuleVersion(Type outputType) {
+    SdkComponent sdkComponent = createComponentWithDefaultMMV("Output type", outputType.getName());
+    if (outputType.asMetadataType() instanceof ArrayType) {
+      for (TypeGeneric typeGeneric : outputType.getGenerics()) {
+        sdkComponent = max(sdkComponent, calculateOutputMinMuleVersion(typeGeneric.getConcreteType()));
+      }
+    } else if (outputType.isSameType(Result.class) || outputType.isSameType(org.mule.sdk.api.runtime.operation.Result.class)) {
+      sdkComponent = max(sdkComponent, getEnforcedMinMuleVersion(outputType));
+      for (TypeGeneric resultTypeGeneric : outputType.getGenerics()) {
+        sdkComponent = max(sdkComponent, getEnforcedMinMuleVersion(resultTypeGeneric.getConcreteType()));
+      }
+    } else if (outputType.isSameType(PagingProvider.class)
+        || outputType.isSameType(org.mule.sdk.api.runtime.streaming.PagingProvider.class)) {
+      sdkComponent = max(sdkComponent, getEnforcedMinMuleVersion(outputType));
+      sdkComponent =
+          max(sdkComponent, getEnforcedMinMuleVersion(outputType.getGenerics().get(1).getConcreteType()));
+    } else {
+      sdkComponent = getEnforcedMinMuleVersion(outputType);
+    }
+    return sdkComponent;
+  }
+
+  private static SdkComponent calculateFieldMinMuleVersion(ExtensionParameter field) {
```

Comment:
none of the error messages generated by this method are helpful to the user. Saying `version V was resolved because X is of type Y` or `X is annotated with Y` is not helpful for the customer.

Messages should be clear, prescriptive and not requiere internal Mule knowledge to interpret. Instead, it should say `version V was resolved because X is annotated with Y, which vas introduced in Mule 4.7.`

Apply this comment to every single error message generated in this class